Add Visibility component documentation (#5326)
mintlify/docs: components/visibility.mdx

---
title: "Visibility"

> description: "Control which content is shown to humans on the web and which is included in markdown output for AI agents."

Reviewer: ```suggestion
description: "Control which content displays in the web UI for human readers and which content displays only in Markdown output for AI agents."
```

keywords: ["visibility", "agents", "humans", "markdown", "AI", "conditional content"]
---

> Use the `Visibility` component to show different content to humans browsing your site versus AI agents consuming the markdown output. This lets you tailor information for each audience without maintaining separate pages.

Reviewer: ```suggestion
Use the `Visibility` component to show different content to humans reading your site versus AI agents processing the Markdown output. This lets you customize information for each audience without maintaining separate pages.
```

> <Tip>Learn more about how Mintlify optimizes your docs for AI tools in [AI-native documentation](/ai-native).</Tip>

Reviewer: ```suggestion
```

Cutting this because it interrupts the flow here and people are already probably pretty motivated to implement this component rather than generally browse about AI native info. Plus it'll probably be auto generated at the bottom of the page.

> On the web, content marked `for="humans"` renders normally and content marked `for="agents"` is hidden. In [markdown page views](/ai-native#page-content) (`.md` URLs), the opposite applies—agent content is included and human-only content is removed.

Reviewer: ```suggestion
On your published site, content marked `for="humans"` renders normally and content marked `for="agents"` is hidden. In [Markdown page views](/ai/markdown-export) (`.md` URLs), the opposite applies—agent content renders in the Markdown output and human-only content does not.
```

> 
> ## Example
> 
> ```mdx
> <Visibility for="humans">
>   Click the **Get started** button in the top-right corner to create your account.
> </Visibility>
> 
> <Visibility for="agents">
>   To create an account, call `POST /v1/accounts` with a valid email address.
> </Visibility>
> ```

Reviewer: ```suggestion
<Visibility for="humans">
  This content displays normally on standard pages of your site.
  
  Use it for information that only humans use like: Click the **Get started** button in the top-right corner to create your account.
</Visibility>

<Visibility for="agents">
  This content only appears on the Markdown version of pages.
  
  Use it for information that only agents use like: To create an account, call `POST /v1/accounts` with a valid email address.
</Visibility>
```

> 

Reviewer: ```suggestion

<Visibility for="humans">
  This content displays normally on standard pages of your site.
  
  Use it for a step like: Click the **Get started** button in the top-right corner to create your account.
</Visibility>

<Visibility for="agents">
  This content only appears on the Markdown version of pages.
  
  Use it for information that only agents use like: To create an account, call `POST /v1/accounts` with a valid email address.
</Visibility>

```

In this example, human readers see the UI-oriented instructions while AI agents receive the API-oriented instructions.

## Properties

<ResponseField name="for" type="string" required>
  The audience for the wrapped content. Must be one of:

>   - `humans` — content is visible on the web and excluded from markdown output

Reviewer: ```suggestion
  - `humans`: Content visible on the web and excluded from Markdown output
```

>   - `agents` — content is hidden on the web and included in markdown output

Reviewer: ```suggestion
  - `agents`: Content hidden on the web and included in Markdown output
```
